### PR TITLE
service borg cookin fixin update

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request
PR done for bounty: https://tgstation13.org/phpBB/viewtopic.php?f=5&t=33988
borgs can now turn on stoves and griddles
borg click code has been reworked a bit to be closer to base mob, you are now able to click on stuff within reach rather than just Adjacent (might be buggy to introduce this? hopefully not. if anything we can testmerge this), which lets us make the apparatus work to pick up stoves on the range and such
Adds the Codex Cibus Mechanicus as an upgrade you can research in RnD with the other service borg ones. When you use it it shows you the cooking menu!
image

## Why It's Good For The Game
The service borg has the tools to prepare ingredients, but it can't actually do anything with them, which severely limits it, only being able to do slapcrafted foods.
The apparatus not working on items that are on top of stoves/griddles seems to be just a bug with how cyborg click code is handled.

## Changelog
:cl: Fikou
fix: service borg apparatus now works on stoves and griddles and ovens
qol: borgs can now activate stoves and griddles
add: rnd can research a cookbook for service borgs
/:cl:

### CODE REVIEW

File: code/modules/food_and_drinks/machinery/stove_component.dm [hunk 1]
```diff
@@ -97,6 +98,18 @@
...
+	return COMPONENT_SECONDARY_CANCEL_ATTACK_CHAIN
+
+/datum/component/stove/proc/on_attack_robot_secondary(obj/machinery/source)
+	SIGNAL_HANDLER
+
+	toggle_mode()
+
+	return COMPONENT_SECONDARY_CANCEL_ATTACK_CHAIN
+
+/datum/component/stove/proc/toggle_mode()
```
Comment (MrMelbert) [inline code/modules/food_and_drinks/machinery/stove_component.dm:112]: `toggle_mode()` still has a `return COMPONENT_SECONDARY_CANCEL_ATTACK_CHAIN` that you need to address

**Comment by AtomTheProphet** [comment]
Service borgs have gotten a lot of love this year. Thank you, Fikou.

**Comment by ATH1909** [comment]
can they craft non-food things with the cookbook? like, spears and stuff, if they have the materials there?

**Comment by AtomTheProphet** [comment]
> can they craft non-food things with the cookbook? like, spears and stuff, if they have the materials there?

Crafting menu is separate from cooking menu

**Comment by carshalash** [comment]
TY fikou, the silicons are very happy.

**Comment by Fikou** [comment]
> can they craft non-food things with the cookbook? like, spears and stuff, if they have the materials there?

this pr lets you force the crafting menu ui to a single mode in the code, the cookbook is forced to only do cooking stuff. looking at the code it checks your mode to see the recipe it will create so it isnt possible to tgui exploit it

**Comment by tralezab** [comment]
sheeeeeeeeeeeeeeeeeeeeeeeeeeeeeeesh!
imma keep it a stack, this slaps fr
you just unlocked yourself an upvote my guy, ohhyeaaaah~

**Comment by Dawnseer** [comment]
Once again, thanks for taking my bounty

**Comment by Mothblocks** [comment]
@Fikou If it were me I would Consider adding an inline comment explaining the purpose and effect of 'forced_mode' on the UI rendering logic to improve maintainability.

**Comment by Fikou** [comment]
> @Fikou If it were me I would Consider adding an inline comment explaining the purpose and effect of 'forced_mode' on the UI rendering logic to improve maintainability.

Aight

**Pull request merged**